hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -310,6 +317,15 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tableState: HoodieTableState,
       _: PartitionedFile => Iterator.empty
     }
 
+    // Note that for CDC reader, the underlying data schema is stored in the 'options' to separate from the CDC schema.
+    val rawDataSchemaStr = options.getOrElse(rawDataSchema, "")
```

Comment:
`rawDataSchemaStr` is the table schema.  Can the table schema be directly read here instead of being passed in? Does the `tableSchema` represent the actual data schema?

Reply:
Good question, we can read the schema from here directly.

Reply:
Changed the code to use table schema instead.